packages/wdio-allure-reporter/src/@types/allure-js-commons.d.ts

```diff
@@ -0,0 +1,100 @@
+// Allure is providing its own types in v2, but there's too many changes,
+// so this is temporarily copied here to customize typings with allure 1
+// Type definitions for allure-js-commons 0.0
+// Project: https://github.com/allure-framework/allure-js
+// Definitions by: Denis Artyuhovich <https://github.com/zaqqaz>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+declare module 'allure-js-commons/beans/step' {
+    declare class Step {}
+    export = Step
+}
+
+declare module 'allure-js-commons' {

```

Comment:
Why do we need this if we have `@types/allure-js-commons` as dependency?

Reply:
Based on the convo above it seems that the typings are not maintained. So I am fine to copy&paste them here but then we should get rid of the `@types/allure-js-commons` dependency.

Reply:
Typescript will throw an error in `test:typings` if there'no `@types/allure-js-commons` packages. But the package alone does not provide enough typings, so I needed to override. I will see what is the minimum overriding needed

Reply:
Please check why the typings test is failing. There should be no need to have `@types/allure-js-commons` if we define it ourselves.

Reply:
> Please check why the typings test is failing. There should be no need to have `@types/allure-js-commons` if we define it ourselves.

You're right, I forgot to mention somehow we're not copying .d.ts files in src folder. So I'll need to modify the build scripts a bit.

Reply:
I can not find any way to ship our defined `allure-js-commons.d.ts`. The problem is even if we copy it in the build folder, Typescript will not recognize the types because: 
- It is not referenced in package.json
- It does not exist in `node_modules/@types`

This is why I needed to include `@types/allure-js-commons` in the first place. It could be used as a work around for Typescript typings.

Reply:
I don't understand. There are two ways how 3rd party module types can be embedded. One is via installing `@types/allure-js-commons` and the other is to declare a module called `allure-js-commons` in a customer types file, e.g. like we do with the [lighthouse module](https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-devtools-service/src/@types/lighthouse.ts#L0-L1) in the devtools-service package. So if we define a custom `allure-js-commons.d.ts` we don't need `@types/allure-js-commons`.

Reply:
Let's either use `@types/allure-js-commons` as dependency or type this package manually.